src/app/resize/resize.component.ts

```diff
@@ -0,0 +1,72 @@
+import {Component, ElementRef, EventEmitter, Input, Output} from "@angular/core";
+
+@Component({
+	selector: 'resize',
+	templateUrl: `
+		<div [class.overlay]="isExpanded"></div>
+		<div
+			class="handle"
+			[class.is-expanded]="isExpanded">
+		</div>
+		`,
+	styles: [`
+	.handle {
+		position: absolute;
+		right: 0;
+		top: 0;
+		height: 100%;
+		width: 15px;
+		border-right: 1px dashed rgba(128, 128, 128, 0.5);
+		cursor: col-resize;
+	}
+
+	.is-expanded {
+		width: 100%;
+	}
+
+	.overlay {
+		position: fixed;
+		top: 0;
+		right: 0;
+		left: 0;
+		bottom: 0;
+		width: 100%;
+		height: 100%;
+	}
+  `],
+	host: {
+		'(mousemove)': 'onMouseMove($event)',
+		'(mousedown)': 'onMouseDown($event)',
+		'(mouseup)': 'onMouseUp()',
+		'(mouseleave)': 'onMouseUp()'
+	}
+})
+export class ResizeComponent {
+	private initOffsetX;
+	private isMouseDown;
+	private isExpanded;
+	private initWidth;
+	@Input('width') width;
+	@Output() widthChange = new EventEmitter();
+
+	constructor(private elRef: ElementRef) {}
+
+	onMouseMove(e) {
+		if (!this.isMouseDown) return
+		this.isExpanded = true;
+
```

Comment:
nit, but trying being more consistent with the spaces around operators, e.g. here:
* Space after `=`
* Space before `-`
* Spaces around `-`
Usually your IDE can do this for you if you reformat.

Same thing with semicolons, either have them everywhere or nowhere (for this codebase - everywhere). 

That's all is very minor, but way easier to read the code when it's consistent.